Tests/Simulation/IntegrationTestFixtureSim.cs

```diff
@@ -95,69 +81,41 @@ public override void SetupTestDevices()
-                        AppSKey = GetAppSessionKey(deviceID),
-                        NwkSKey = GetNetworkSessionKey(deviceID),
-                        DevAddr = DevAddr.Parse(deviceID.ToString("00000000", CultureInfo.InvariantCulture)),
-                    });
-            }
+            for (var deviceId = 1100; deviceId <= 1110; deviceId++)
+                this.deviceRange1000_ABP.Add(CreateAbpDevice(deviceId));
 
-            for (var deviceId = 4000; deviceId < 4000 + Configuration.NumberOfLoadTestDevices; deviceId++)
-            {
-                this.deviceRange4000_OTAA_FullLoad.Add(
-                    new TestDeviceInfo
-                    {
-                        DeviceID = deviceId.ToString("0000000000000000", CultureInfo.InvariantCulture),
-                        AppEui = JoinEui.Parse(deviceId.ToString("0000000000000000", CultureInfo.InvariantCulture)),
-                        AppKey = GetAppKey(deviceId),
-                        IsIoTHubDevice = true,
-                        SensorDecoder = "DecoderValueSensor",
-                    });
-            }
+            for (var deviceId = 2000; deviceId < 2000 + Configuration.NumberOfLoadTestDevices; deviceId++)
+                this.deviceRange2000_ABP_FullLoad.Add(CreateAbpDevice(deviceId));
 
-            /*
-            // Range of 1000 ABP devices from 2000 to 2999: Used for load testing
-            for (var deviceID = 2000; deviceID <= 2999; deviceID++)
-            {
-                this.deviceRange2000_1000_ABP.Add(
-                    new TestDeviceInfo
-                    {
-                        DeviceID = deviceID.ToString("0000000000000000", CultureInfo.InvariantCulture),
-                        GatewayID = gatewayID,
-                        IsIoTHubDevice = true,
-                        SensorDecoder = "DecoderValueSensor",
-                        KeepAliveTimeout = 0,
-                        AppSKey = GetAppSessionKey(deviceID),
-                        NwkSKey = GetNetworkSessionKey(deviceID),
-                        DevAddr = DevAddr.Parse(deviceID.ToString("00000000", CultureInfo.InvariantCulture)),
-                    });
-            }
+            for (var deviceId = 3000; deviceId < 3000 + Configuration.NumberOfLoadTestDevices; deviceId++)
+                this.deviceRange3000_OTAA_FullLoad.Add(CreateOtaaDevice(deviceId));
 
-            // Range of 10 OTAA devices from 3000 to 3009: Used for load testing
-            for (var deviceID = 3000; deviceID <= 3009; deviceID++)
-            {
-                this.deviceRange3000_10_OTAA.Add(
-                    new TestDeviceInfo
-                    {
-                        DeviceID = deviceID.ToString("0000000000000000", CultureInfo.InvariantCulture),
-                        AppEui = JoinEui.Parse(deviceID.ToString("0000000000000000", CultureInfo.InvariantCulture)),
-                        AppKey = GetAppKey(deviceID),
-                        GatewayID = gatewayID,
```

Comment:
Not sure if it's an issue but with the new helper method `CreateOtaaDevice` no GatewayId is assigned.

Reply:
Eagle eyes :) It was intentional though, since for the multi-gateway scenarios we do not want to associate these devices with a gateway already.